core/src/main/java/com/linecorp/armeria/server/DefaultServiceRequestContext.java

```diff
@@ -456,12 +456,20 @@ public void setRequestTimeoutAfter(Duration requestTimeout) {
     public void setRequestTimeoutAtMillis(long requestTimeoutAtMillis) {
         checkArgument(requestTimeoutAtMillis >= 0,
                       "requestTimeoutAtMillis: " + requestTimeoutAtMillis + " (expected: >= 0)");
-        final long nowMillis = Instant.now().toEpochMilli();
-        final long requestTimeoutAfter = requestTimeoutAtMillis - nowMillis;
-        checkArgument(requestTimeoutAfter > 0,
-                      "requestTimeoutAtMillis: %s (expected: > 'now=%s')", requestTimeoutAtMillis, nowMillis);
-
-        setRequestTimeoutAfterMillis(requestTimeoutAfter);
+        final long requestTimeoutAfter = requestTimeoutAtMillis - System.currentTimeMillis();
+
+        if (requestTimeoutAfter <= 0) {
+            final TimeoutController requestTimeoutController = this.requestTimeoutController;
+            if (requestTimeoutController != null) {
+                if (eventLoop().inEventLoop()) {
+                    requestTimeoutController.timeoutNow();
+                } else {
+                    eventLoop().execute(requestTimeoutController::timeoutNow);
+                }
+            }
```

Comment:
Don't we need to add `pendingTimeoutTask` here?

Reply:
Indeed! Updated. 😀